pkg/parse/parse.go

```diff
@@ -607,29 +605,35 @@ func GetBuildOutput(buildOutput string) (define.BuildOutputOption, error) {
 				return define.BuildOutputOption{}, fmt.Errorf("duplicate %q not supported", arr[0])
 			}
 			typeSelected = true
-			if arr[1] == "local" {
-				isDir = true
-			} else if arr[1] == "tar" {
-				isDir = false
-			} else {
+			switch arr[1] {
+			case define.Local.String():
+				out.IsDir = true
+			case define.Tar.String():
+				out.IsDir = false
+			case define.Registry.String():
+				out.Type = define.Image
```

Comment:
`case Registry: Type = Image`???

It’s fine to have the UI strings be different from the internal implementation enum names. It’s quite hard to maintain code that has a single enum that contains both internal “type” values and “strings”, with some values usable as both, but some only usable as one of the two.

Reply:
> The registry exporter is a shortcut for type=image,push=true.

Ah, I can see how that's misleading but I tried to do this conversion here itself rather "implicitly". I'll take a look at this. I guess this is what you meant?

Reply:
Basically I think it would be easiest if the the enum should were reserved for the `out.Type` fields, and this parser would just hard-code the specific string values in the `switch`.

(See elsewhere about buildx specifics, and unimplemented features, on the CLI layer vs. internal representation.)